stdlib/public/core/CollectionAlgorithms.swift

```diff
@@ -209,6 +209,84 @@ extension BidirectionalCollection where Element: Equatable {
   }
 }
 
+//===----------------------------------------------------------------------===//
+// indices(where:) / indices(of:)
+//===----------------------------------------------------------------------===//
+
+#if !$Embedded
+extension Collection {
+  /// Returns the indices of all the elements that match the given predicate.
+  ///
+  /// For example, you can use this method to find all the places that a
+  /// vowel occurs in a string.
+  ///
+  ///     let str = "Fresh cheese in a breeze"
+  ///     let vowels: Set<Character> = ["a", "e", "i", "o", "u"]
+  ///     let allTheVowels = str.indices(where: { vowels.contains($0) })
+  ///     // str[allTheVowels].count == 9
+  ///
+  /// - Parameter predicate: A closure that takes an element as its argument
+  ///   and returns a Boolean value that indicates whether the passed element
+  ///   represents a match.
+  /// - Returns: A set of the indices of the elements for which `predicate`
+  ///   returns `true`.
+  ///
+  /// - Complexity: O(*n*), where *n* is the length of the collection.
+  @available(SwiftStdlib 5.11, *)
+  public func indices(where predicate: (Element) throws -> Bool) rethrows
+    -> RangeSet<Index>
+  {
+    if isEmpty { return RangeSet() }
```

Comment:
I don't think that you have to special-case this; it falls out from the rest of the implementation for free, doesn't it?

Reply:
Good point, I think this was a leftover piece from when `RangeSet` had a special-case representation for the empty set so this ensured we hit that fast path. Will fix